src/main/java/org/opensearch/security/http/SecurityNonSslHttpServerTransport.java

```diff
@@ -82,4 +89,9 @@ protected void initChannel(Channel ch) throws Exception {
             super.initChannel(ch);
         }
     }
+
+    @Override
+    protected ChannelInboundHandlerAdapter createHeaderVerifier() {
```

Comment:
Can you brief comment on the role of this method in request handling?

Reply:
This is the new extension point. It allows a NetworkPlugin to define a HeaderVerifier which is already inserted before decompression in the netty pipeline in the companion PR.

Reply:
This class is a subclass of Netty4HttpServerTransport, a class defined in core